econml/drlearner.py

```diff
@@ -656,9 +660,228 @@ def multitask_model_cate(self):
+
+    n_splits: int, cross-validation generator or an iterable, optional (Default=2)
+        Determines the cross-validation splitting strategy.
+        Possible inputs for cv are:
+
+        - None, to use the default 3-fold cross-validation,
+        - integer, to specify the number of folds.
+        - :term:`CV splitter`
+        - An iterable yielding (train, test) splits as arrays of indices.
+
+        For integer/None inputs, if the treatment is discrete
+        :class:`~sklearn.model_selection.StratifiedKFold` is used, else,
+        :class:`~sklearn.model_selection.KFold` is used
+        (with a random shuffle in either case).
+
+        Unless an iterable is used, we call `split(X,T)` to generate the splits.
+
+    random_state: int, :class:`~numpy.random.mtrand.RandomState` instance or None
+        If int, random_state is the seed used by the random number generator;
+        If :class:`~numpy.random.mtrand.RandomState` instance, random_state is the random number generator;
+        If None, the random number generator is the :class:`~numpy.random.mtrand.RandomState` instance used
+        by :mod:`np.random<numpy.random>`.
+
+    Examples
+    --------
+    A simple example with the default models::
+
+        import numpy as np
+        import scipy.special
+        from econml.drlearner import DRLearner, SparseLinearDRLearner
+
+        np.random.seed(123)
+        X = np.random.normal(size=(1000, 3))
+        T = np.random.binomial(2, scipy.special.expit(X[:, 0]))
+        y = (1 + .5*X[:, 0]) * T + X[:, 0] + np.random.normal(size=(1000,))
+        est = SparseLinearDRLearner()
+        est.fit(y, T, X=X, W=None, inference='debiasedlasso')
+
+    >>> est.effect(X[:3])
+    array([ 0.46089994,  0.31993838, -0.07386204])
+    >>> est.effect_interval(X[:3])
+    (array([ 0.11912103, -0.16476616, -0.64903889]),
+     array([0.80267886, 0.80464292, 0.50131482]))
+    >>> est.coef_(T=1)
+    array([0.40984866, 0.02624624, 0.05320565])
+    >>> est.coef__interval(T=1)
+    (array([ 0.21365131, -0.15865431, -0.13733605]),
+     array([0.606046  , 0.21114678, 0.24374735]))
+    >>> est.intercept_(T=1)
+    0.864611566994208
+    >>> est.intercept__interval(T=1)
+    (0.6779174404370922, 1.0513056935513236)
+
+    Attributes
+    ----------
+    score_ : float
+        The MSE in the final doubly robust potential outcome regressions, i.e.
+
+        .. math::
+            \\frac{1}{n_t} \\sum_{t=1}^{n_t} \\frac{1}{n} \\sum_{i=1}^n (Y_{i, t}^{DR} - \\hat{\\theta}_t(X_i))^2
+
+        where n_t is the number of treatments (excluding control).
+
+        If `sample_weight` is not None at fit time, then a weighted average across samples is returned.
+
+    """
+
+    def __init__(self,
+                 model_propensity=LogisticRegressionCV(cv=3, solver='lbfgs', multi_class='auto'),
+                 model_regression=WeightedLassoCV(cv=3),
+                 featurizer=None,
+                 fit_cate_intercept=True,

```

Comment:
Is this the case? I remember we had a conversation about this and you said that the cate intercept is not the same as the final model intercept. For the SparseDML estimator, fit_intercept is always false in the final model.

Reply:
Yeap, here it should be true. This is the cate_intercept (i.e. if cate is <theta, phi> + beta or just <theta, phi>), not the final model intercept in the DML (which is a weird thing that has nothing to do with the CATE intercept). It's true that there is a small discrepancy in our interface between DML and DR. Keith is fixing that discrepancy and we will have fit_cate_intercept in DML too, which will be set to True by default.